gax/src/main/java/com/google/api/gax/core/Distribution.java

```diff
@@ -31,128 +31,75 @@
 
 import com.google.api.core.BetaApi;
 import com.google.common.base.MoreObjects;
-import com.google.common.base.MoreObjects.ToStringHelper;
 import com.google.common.base.Preconditions;
-import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLongArray;
 
 /**
- * Takes measurements and stores them in linear buckets from 0 to totalBuckets - 1, along with
- * utilities to calculate percentiles for analysis of results.
+ * Distribution records values from {@code 0} (inclusive) to {@code maxValue} (exclusive) and
```

Comment:
@garrettjonesgoogle Do you have a better name for `maxValue`? It's exclusive so the name is a little misleading.

Reply:
Maybe `endValue`? (like in C++, where end() on a collection type is actually past the last element)

Reply:
Good idea. Done